server/src/main/java/org/opensearch/index/translog/RemoteFsTimestampAwareTranslog.java

```diff
@@ -240,7 +261,12 @@ public void onResponse(List<BlobMetadata> blobMetadata) {
                         remoteGenerationDeletionPermits.release();
                     }
                 } catch (Exception e) {
+                    logger.error("Exception in trimUnreferencedReaders", e);
                     remoteGenerationDeletionPermits.release(REMOTE_DELETION_PERMITS);
```

Comment:
When can there be a case now that we would need to release 2 permits during exception thrown now?

Reply:
This is just a catch-all box for any exception that occurs in the try block.